aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -0,0 +1,169 @@
+ * native fun try_sub(aggregator: &mut Aggregator, value: u128): bool;
+ **************************************************************************************************/
+fn native_try_sub(
+    context: &mut SafeNativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert_eq!(args.len(), 2);
+
+    context.charge(AGGREGATOR_V2_TRY_SUB_BASE)?;
+
+    // Get aggregator information and a value to subtract.
+    let value = safely_pop_arg!(args, u128);
+    let (id, limit) = aggregator_info(&safely_pop_arg!(args, StructRef))?;
+
+    // Get aggregator.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+    let aggregator = aggregator_data.get_aggregator(id, limit)?;
+    Ok(smallvec![Value::bool(aggregator.try_sub(value).is_ok())])
+}
+
+/***************************************************************************************************
+ * native fun read(aggregator: &Aggregator): u128;
+ **************************************************************************************************/
+
+fn native_read(
+    context: &mut SafeNativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert_eq!(args.len(), 1);
+
+    context.charge(AGGREGATOR_V2_READ_BASE)?;
+
+    // Extract information from aggregator struct reference.
+    let (id, limit) = aggregator_info(&safely_pop_arg!(args, StructRef))?;
+
+    // Get aggregator.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+    let aggregator = aggregator_data.get_aggregator(id, limit)?;
+
+    let value = aggregator.read_and_materialize(aggregator_context.resolver, &id)?;
+
+    Ok(smallvec![Value::u128(value)])
+}
+
+/***************************************************************************************************
+ * native fun destroy(aggregator: Aggregator);
+ **************************************************************************************************/
+
+fn native_destroy(
+    context: &mut SafeNativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert_eq!(args.len(), 1);
+
+    context.charge(AGGREGATOR_V2_DESTROY_BASE)?;
+
+    // Extract information from aggregator struct reference.
+    let (id, _limit) = aggregator_info(&safely_pop_arg!(args, StructRef))?;
+
+    // Get aggregator data.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+
+    // Actually remove the aggregator.
+    aggregator_data.remove_aggregator(id);
```

Comment:
just to check, will this work when AggregatorSnapshot is in the picture? i.e. do we need anything from the aggregator that was cloned from?